reference-implementation/lib/transform-stream.js

```diff
@@ -45,10 +47,16 @@ function TransformStreamEnqueueToReadable(transformStream, chunk) {
     throw transformStream._storedError;
   }
 
-  const backpressure = controller.desiredSize <= 0;
+  const maybeBackpressure = controller.desiredSize <= 0;
 
-  if (transformStream._readableBackpressure !== backpressure) {
-    TransformStreamSetBackpressure(transformStream, backpressure);
+  if (maybeBackpressure === true && transformStream._backpressure === false) {
+    // This allows pull() again. When desiredSize is 0, it's possible that a pull() is made immediately (but
+    // asynchronously) after this because of pending read()s and fix _backpressure back to false.
+    //
+    // We don't have to worry about that such a pull() may happen synchronously to the enqueue(), and therefore is not
```

Comment:
Sorry, I don't understand what "synchronously to" means here. Maybe you could explain the dangerous case as a series of steps, and then explain why it never happens?